examples/FileDialog/Readme.md

```diff
@@ -1,15 +1,33 @@
-You can programmatically invoke default OS file prompt. In order to do that you'll have to set the ref on your `Dropzone` instance and call the instance `open` method.
-
 **Please note**, that for security reasons most browsers require popups and dialogues to originate from a direct user interaction (i.e. click). If you are calling `dropzoneRef.open()` asynchronously, there’s a good chance it’s going to be blocked by the browser. So if you are calling `dropzoneRef.open()` asynchronously, be sure there is no more than *1000ms* delay between user interaction and `dropzoneRef.open()` call. Due to the lack of official docs on this (at least we haven’t found any. If you know one, feel free to open PR), there is no guarantee that **allowed delay duration** will not be changed in later browser versions. Since implementations may differ between different browsers, avoid calling open asynchronously if possible.
 
+You can programmatically invoke default OS file prompt. There are two ways to do that. The first is to provide a function as the child for `Dropzone` to access the `open` method as a parameter. The second way is to set the ref on your `Dropzone` instance and call the instance `open` method.
+
+##### Open programmatically by Child Function
+
+```
+<Dropzone onDrop={files => alert(JSON.stringify(files))} disableClick>
+  {({ open }) => (
+    <React.Fragment>
+      <button type="button" onClick={() => open()}>
+        Open File Dialog
+      </button>
+
+      <p>Drop files here.</p>
+    </React.Fragment>
+  )}
+</Dropzone>
+```
+
+##### Open programmatically by Ref
+
 ```
-let dropzoneRef;
+const dropzoneRef = React.createRef();
```

Comment:
We still need to keep the option to invoke `.open()` using a ref as some users may not use the render fn.

So keep the old example and add another to illustrate what you have added.

Reply:
This is the ref example, I updated it to use the modern `React.createRef()` API instead of the callback API (to encourage best practices) but the callback API still works. Are you suggesting they be split up into two different files?

Reply:
No, I just missed that, it's been a long day 😄